etc/testing/circle/workloads/go-test-results/egress/main.go

```diff
@@ -0,0 +1,194 @@
+	}
+	var jobInfo gotestresults.JobInfo
+	if err = json.Unmarshal(jobInfoJson, &jobInfo); err != nil {
+		return nil, errors.Wrapf(err, "unmarshalling job info json ")
+	}
+	return &jobInfo, nil
+}
+
+func insertJobInfo(jobInfo *gotestresults.JobInfo, jobInfoPaths map[string]gotestresults.JobInfo, path string, d fs.DirEntry, db *sqlx.DB) error {
+	_, err := db.Exec(`INSERT INTO public.ci_jobs (
+							workflow_id, 
+							job_id, 
+							job_executor,
+							job_name, 
+							job_timestamp, 
+							job_num_executors,
+							commit_sha, 
+							branch, 
+							tag,
+							pull_requests
+						) VALUES ($1, $2, $3, $4, $5, $6, $7, $8, $9, $10)`,
+		jobInfo.WorkflowId,
+		jobInfo.JobId,
+		jobInfo.JobExecutor,
+		jobInfo.JobName,
+		jobInfo.JobTimestamp,
+		jobInfo.JobNumExecutors,
+		jobInfo.Commit,
+		jobInfo.Branch,
+		jobInfo.Tag,
+		jobInfo.PullRequests,
+	)
+	if err != nil {
+		var pgErr *pgconn.PgError
+		if errors.As(err, &pgErr) && pgErr.Code == "23505" {
+			// Unique key constraint violation. In other words, the record already exists.
+			// Ignore insert to since we don't want to change the timestamp, but log it. This is expected with multiple executors.
+			logger.Printf("Job info already inserted, Skipping insert for workflow %v  job %v  executor %v\n", jobInfo.WorkflowId, jobInfo.JobId, jobInfo.JobExecutor)
+			return nil
+		}
+		return errors.Wrapf(err, "inserting job info for workflow %v  job %v  executor %v", jobInfo.WorkflowId, jobInfo.JobId, jobInfo.JobExecutor)
+	}
+	jobInfoPaths[strings.TrimSuffix(path, d.Name())] = *jobInfo // read for use when inserting individual test results, don't insert if errored
+	logger.Printf("Inserted job info for workflow %v job %v executor %v", jobInfo.WorkflowId, jobInfo.JobId, jobInfo.JobExecutor)
+	return nil
+}
+
+func insertTestResultFile(path string, jobInfoPaths map[string]gotestresults.JobInfo, db *sqlx.DB) error {
+	resultsFile, err := os.Open(path)
+	if err != nil {
+		return errors.Wrapf(err, "opening results file ")
+	}
+	defer resultsFile.Close()
+	resultsScanner := bufio.NewScanner(resultsFile)
+	for resultsScanner.Scan() {
+		resultJson := resultsScanner.Bytes()
+		var result TestResultLine
+		err = json.Unmarshal(resultJson, &result)
+		if err != nil {
+			logger.Printf("Failed unmarshalling file as a test result - %s - %v. Continuing to parse results file.\n", path, err)
+			continue
+		}
+		// output is mostly logs and coverage and it's not that useful, so we don't store output-only actions
+		if result.Action != "output" {
+			splitPath := strings.Split(path, "/")
+			fileName := splitPath[len(splitPath)-1]
+			// get job info previously inserted in order to link foreign keys to jobs table
+			jobInfo, ok := jobInfoPaths[strings.TrimSuffix(path, fileName)]
+			if !ok {
+				return errors.WithStack(fmt.Errorf("Failed to find job info for %v - file name %v - job infos %v ", path, fileName, jobInfoPaths))
+			}
+			_, err = db.Exec(`INSERT INTO public.test_results (
```

Comment:
BTW, there is a NamedExec method where you can use :go_name: instead of $1 placeholders, and then just pass a struct instead of individual args.

Reply:
I changed this for ci_jobs table, but since test-results pulls from 2 separate structs NamedExec didn't seem as helpful.